Add swift-bin to library path

iKenWoo opened the pull request:
Adding `.swift-bin` to the library build path so that dynamic libraries built by swift package manager can be loaded.

This will resolve #7.

briancroom commented:
Alternatively, perhaps the step that copies the built products should be smarter, and be sure to put shared libraries into .swift-lib, where they really belong?

kylef commented:
I agree with @briancroom, the shared libraries belong in the lib folder and not bin. We should copy the files from `.build/release` with the file extension `.so` into the lib directory.

As a separate step, we can make it so that only executable files are copied into the bin directory, so only the binaries should end up there.

@iKenWoo Would you be willing to update the PR so that we instead copy the .so files into the lib directory?

iKenWoo commented:
sure i'll make that change

iKenWoo commented:
Changed it so that `.so` files in `.build/release` are copied to `.swift-lib` and to exclude `.so` files when copying executable files into `swift-bin`.

Also removed `swift-bin` from the library path.

kylef commented:
Perfect, thanks @iKenWoo
